09-features.md

```diff
@@ -41,6 +41,7 @@ The Context column decodes as follows:
 | 20/21 | `option_anchor_outputs`          | Anchor outputs                                            | IN       | `option_static_remotekey` | [BOLT #3](03-transactions.md)         |
 | 22/23 | `option_anchors_zero_fee_htlc_tx` | Anchor commitment type with zero fee HTLC transactions   | IN       | `option_static_remotekey` | [BOLT #3][bolt03-htlc-tx], [lightning-dev][ml-sighash-single-harmful]|
 | 26/27 | `option_shutdown_anysegwit`         | Future segwit versions allowed in `shutdown`              | IN       |                   | [BOLT #2][bolt02-shutdown]   |
+| 28/29 | `option_dual_fund`             | Use v2 of channel open, enables dual funding              | IN      | `option_static_remotekey`   | [BOLT #2](02-peer-protocol.md)        |
```

Comment:
There is no logical dependency on `option_static_remotekey` for dual-funding, why force one? If in the future we move to a new commitment format that doesn't depend on `option_static_remotekey` and is compatible with dual funding and superior to the existing commitment formats, we want nodes to be able to continue using dual funding with that new commitment format without having to activate a legacy `option_static_remotekey` feature.

```suggestion
| 28/29 | `option_dual_fund`             | Use v2 of channel open, enables dual funding              | IN      |                             | [BOLT #2](02-peer-protocol.md)        |
```

Reply:
I think we agree about this, and I've got some misunderstanding wrt how best to signal that in features.

The desired behavior is that `static remotekeys are assumed` for current commitment opens with v2, so that we can eventually deprecate/remove the option entirely (all channel opens using the current commitment scheme use it).

I thought that we'd get this by signaling that it's a dependency; is there a better way to just assume it though?

Basically I want to enforce "current commitment formats that are using v2 MUST use the static remotekey".

Reply:
> Basically I want to enforce "current commitment formats that are using v2 MUST use the static remotekey".

I think that this is something that you do at the node level: if that's what you want, your node should signal `option_static_remotekey:mandatory` in its `init` and `node_announcement`, but that is independent of dual funding. I believe all implementations will eventually start doing this to be able to deprecate pre-static_remotekey commitments, but that really is independent of the dual funding specification, isn't it?

Otherwise we're really painting ourselves into a corner when in the future we'll want to keep using dual funding but will want to deprecate plain `option_static_remotekey` (e.g. in favor of an updated version of anchor outputs). FWIW, it was a mistake to make `option_static_remotekey` a dependency of `anchor_outputs`, for exactly the same reason: nodes are unable today to say "I only want anchor outputs channels, don't try to open a static_remotekey channel that doesn't use anchor outputs".